tls/s2n_alerts.h

```diff
@@ -23,3 +23,4 @@ extern int s2n_process_alert_fragment(struct s2n_connection *conn);
 extern int s2n_queue_writer_close_alert_warning(struct s2n_connection *conn);
 extern int s2n_queue_reader_unsupported_protocol_version_alert(struct s2n_connection *conn);
 extern int s2n_queue_reader_handshake_failure_alert(struct s2n_connection *conn);
+extern int s2n_queue_reader_illegal_parameter_alert(struct s2n_connection *conn);
```

Comment:
By default (and counter to the RFC) we don't actually send specific alerts. We just close the connection. There have been issues in the past with specific alerts giving attackers too much information.

Is there a reason to send a specific alert here?

Reply:
The RFC calls for it. I like having the alert because it helps determine why the hello_recv fails (downgrade attack or some other issue). We could add a specific error code instead of the alert, but I'd like to be sure the test catches the actual downgrade attack instead of accidentally failing for some other reason.

Reply:
Can we go the specific error code route? I would say testing isn't a good enough argument for a new alert / deviating from our standard practices. You should also be able to verify the correct error in your tests.

Reply:
Updated in next push.